### ISSUE TITLE
jwt header missing kid, for google service accounts?

### ISSUE THREAD

**Issue by macMikey** [4d/4D-NetKit#70]
[google service account docs for the jwt header](https://developers.google.com/identity/protocols/oauth2/service-account) includes a header **kid**, which is the private key id.

**Comment by e-marchand** [comment]
I see in code here a jwt

https://github.com/4d/4D-NetKit/blob/a77462e4704ec889fe8eee884e735fe8232b57db/Project/Sources/Classes/OAuth2Provider.4dm#L543

So the feature could be as I understand, allow to add it?

**Comment by macMikey** [comment]
right. i have oauth2 working with service accounts with my 4d google lib. i have **kid** implemented in my **JWT** generator (but that doesn't mean that it can't work without it. stay tuned, still testing.

**Comment by macMikey** [comment]
this goes along with #71. see [this comment](https://github.com/4d/4D-NetKit/issues/71#issuecomment-2192095403) for how i propose implementing this.
